File: MiPrimeraAplicacion/app/src/main/java/com/micky/miprimeraaplicacion/UpdateActivity.java
package com.micky.miprimeraaplicacion;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.micky.miprimeraaplicacion.backend.BaseDatos;

public class UpdateActivity extends AppCompatActivity {
    private EditText editTextPaterno;
    private EditText editTextMaterno;
    private EditText editTextNombre;
    private Button buttonActualizar;
    private int personaId;
    private BaseDatos baseDatos;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_update);

        // Inicializar la base de datos
        baseDatos = new BaseDatos(this, "nombre_de_la_base_de_datos", null, 1);

        // Obtener los datos de la persona a editar del Intent
        Bundle extras = getIntent().getExtras();
        if (extras != null) {
            personaId = extras.getInt("personaId");
            String paterno = extras.getString("paterno");
            String materno = extras.getString("materno");
            String nombre = extras.getString("nombre");

            // Referencias a los elementos de la interfaz
            editTextPaterno = findViewById(R.id.editTextPaterno);
            editTextMaterno = findViewById(R.id.editTextMaterno);
            editTextNombre = findViewById(R.id.editTextNombre);
            buttonActualizar = findViewById(R.id.buttonActualizar);

            // Mostrar los datos de la persona en los EditText
            editTextPaterno.setText(paterno);
            editTextMaterno.setText(materno);
            editTextNombre.setText(nombre);
        }

        // Configurar el botón para actualizar los datos de la persona
        buttonActualizar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                actualizarPersona();
            }
        });
    }

    // Método para actualizar los datos de la persona en la base de datos
    private void actualizarPersona() {
        String paterno = editTextPaterno.getText().toString().trim();
        String materno = editTextMaterno.getText().toString().trim();
        String nombre = editTextNombre.getText().toString().trim();

        if (!paterno.isEmpty() && !nombre.isEmpty()) {
            baseDatos.actualizarPersona(personaId, paterno, materno, nombre);
            Toast.makeText(this, "Persona actualizada correctamente", Toast.LENGTH_SHORT).show();
            finish(); // Cerrar la actividad y volver atrás
        } else {
            Toast.makeText(this, "Por favor ingresa al menos el paterno y nombre", Toast.LENGTH_SHORT).show();
        }
    }
}

File: PrimerKotlin/app/src/main/java/com/micky/primerkotlin/UpdateActivity.kt
package com.micky.primerkotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.micky.primerkotlin.backend.BaseDatos
import com.micky.primerkotlin.backend.Lugares

class UpdateActivity : AppCompatActivity() {
    private lateinit var etNombre: EditText
    private lateinit var etDescripcion: EditText
    private lateinit var etLatitud: EditText
    private lateinit var etLongitud: EditText
    private lateinit var btnActualizar: Button
    private lateinit var btnVolver: Button
    private lateinit var dbHandler: BaseDatos

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_update)

        etNombre = findViewById(R.id.etNombre)
        etDescripcion = findViewById(R.id.etDescripcion)
        etLatitud = findViewById(R.id.etLatitud)
        etLongitud = findViewById(R.id.etLongitud)
        btnActualizar = findViewById(R.id.btnActualizar)
        btnVolver = findViewById(R.id.btnVolver)
        dbHandler = BaseDatos(this)

        val lugarId = intent.getIntExtra("lugar_id", -1)
        if (lugarId != -1) {
            val lugar = dbHandler.getLugar(lugarId)
            etNombre.setText(lugar.nombre)
            etDescripcion.setText(lugar.descripcion)
            etLatitud.setText(lugar.latitud.toString())
            etLongitud.setText(lugar.longitud.toString())
        }

        btnActualizar.setOnClickListener {
            val nombre = etNombre.text.toString()
            val descripcion = etDescripcion.text.toString()
            val latitud = etLatitud.text.toString().toFloatOrNull()
            val longitud = etLongitud.text.toString().toFloatOrNull()

            if (latitud == null || longitud == null) {
                Toast.makeText(this, "Ingrese valores válidos para latitud y longitud", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            val lugar = Lugares()
            lugar.id = lugarId
            lugar.nombre = nombre
            lugar.descripcion = descripcion
            lugar.latitud = latitud
            lugar.longitud = longitud

            val success = dbHandler.updateLugar(lugar)
            if (success) {
                Toast.makeText(this, "Lugar actualizado correctamente", Toast.LENGTH_SHORT).show()
                setResult(RESULT_OK)
                finish()
            } else {
                Toast.makeText(this, "Error al actualizar el lugar", Toast.LENGTH_SHORT).show()
            }
        }

        btnVolver.setOnClickListener {
            finish()
        }
    }
}
